### PR TITLE
update locations, sync server redirects

### PR DESCRIPTION
**Pull request by ryansolid**
This PR breaks apart "/components" and moves stuff around. Introduces no render entries.

### CODE REVIEW

File: packages/start/entry-server/entries.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { renderToStringAsync, renderToStream } from "solid-js/web";
+import { RequestContext } from "./StartServer";
+
+export function renderAsync(fn: (context: RequestContext) => any, options?: any) {
```
Comment (nksaraf) [inline packages/start/entry-server/entries.ts:4]: Should probably type this as JSX.Element

Comment (nksaraf) [reply packages/start/entry-server/entries.ts:4]: And options as:
```
{
    timeoutMs?: number;
    nonce?: string;
    renderId?: string;
}
```

File: packages/start/entry-server/entries.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    context.responseHeaders.set("content-type", "text/html");
+
+    return new Response(markup, {
+      status: 200,
+      headers: context.responseHeaders
+    });
+  };
+}
+
+function handleRedirect(context) {
```
Comment (nksaraf) [inline packages/start/entry-server/entries.ts:20]: Is this supposed to be unimplemented?

File: packages/start/entry-client/StartClient.tsx [hunk 3]
```diff
@@ -1,21 +1,16 @@
...
-const [request, setRequest] = createSignal(new Request("http://localhost:3000/"));
 
 export default () => {
   return (
     <StartProvider
-      context={{
-        responseHeaders: new Headers(),
-        request: request()
-      }}
+      context={{}}
```
Comment (nksaraf) [inline packages/start/entry-client/StartClient.tsx:13]: This has typescript issues, maybe mark as any or @ts-expect-error, but I think we should stub the values out with reasonable defaults. Though probably even better to make it a proxy and throw on the Client if the user tries to access it on the client.

File: packages/start/entry-client/StartClient.tsx [hunk 4]
```diff
@@ -1,21 +1,16 @@
 import { MetaProvider } from "solid-meta";
 import { Router } from "solid-app-router";
-import { StartProvider } from "./StartContext";
+import { StartProvider } from "../entry-server/StartContext";
 // @ts-expect-error
```
Comment (nksaraf) [inline packages/start/entry-client/StartClient.tsx:4]: this can now be deleted, should have done it earlier but yeah can be removed now.

**Review by nksaraf — COMMENTED** [review]
a few fixes here and there, I can clean it up while merging if you want

File: packages/start/entry-server/entries.ts (hunk 2 above)
Comment (ryansolid) [reply packages/start/entry-server/entries.ts:20]: It was causing infinite loops. So I left it for now.

**Pull request merged**